### PR TITLE
Reverted Xtext 2.8.4 upgrade

### PR DESCRIPTION
**Pull request by kaikreuzer**
Signed-off-by: Kai Kreuzer <kai@openhab.org>

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -15,7 +15,3 @@ distribution/smarthome/runtime/etc/mapdb/storage.mapdb.t
 bundles/storage/org.eclipse.smarthome.storage.mapdb.test/userdata/mapdb/storage.mapdb
 bundles/storage/org.eclipse.smarthome.storage.mapdb.test/userdata/mapdb/storage.mapdb.p
 bundles/storage/org.eclipse.smarthome.storage.mapdb.test/userdata/mapdb/storage.mapdb.t
-bundles/model/antlr-generator-3.2.0-patch.jar
```
Comment (maggu2810) [inline .gitignore]: Has the location of antlr-generator and the storage.mapdb been changed by the Xtext update?

File: targetplatform/EclipseSmartHome.setup [hunk 2]
```diff
@@ -2277,16 +2277,16 @@
           name="org.eclipse.m2e.logback.feature.feature.group"/>
       <requirement
           name="org.eclipse.xtend.sdk.feature.group"
-          versionRange="[2.8.0,2.8.4]"/>
+          versionRange="[2.6.0,2.7.0]"/>
```
Comment (maggu2810) [inline targetplatform/EclipseSmartHome.setup]: For this and the three diffs below we should use [x,y)

**Comment by maggu2810** [comment]
Is the setup of your IDE working?

```text
Executing startup tasks
Product org.eclipse.products.epp.package.java.latest.released
Project org.eclipse.smarthome.core.master
Bundle org.eclipse.oomph.setup 1.3.0.v20151119-0756, build=1955, branch=1c21883e4012ea9ca63d5ee8fc0d9a02fc7b4d75, commit=1c21883e4012ea9ca63d5ee8fc0d9a02fc7b4d75
Bundle org.eclipse.oomph.setup.core 1.3.0.v20151119-1107, build=1955, branch=1c21883e4012ea9ca63d5ee8fc0d9a02fc7b4d75, commit=1c21883e4012ea9ca63d5ee8fc0d9a02fc7b4d75
Bundle org.eclipse.oomph.setup.git 1.3.0.v20151017-1318, build=1923, branch=8359c49a92734051d767cf7c2d0b8c8d16961b51, commit=8359c49a92734051d767cf7c2d0b8c8d16961b51
Bundle org.eclipse.oomph.setup.jdt 1.3.0.v20150923-0945, build=1923, branch=8359c49a92734051d767cf7c2d0b8c8d16961b51, commit=8359c49a92734051d767cf7c2d0b8c8d16961b51
Bundle org.eclipse.oomph.setup.launching 1.3.0.v20150923-0945, build=1923, branch=8359c49a92734051d767cf7c2d0b8c8d16961b51, commit=8359c49a92734051d767cf7c2d0b8c8d16961b51
Bundle org.eclipse.oomph.setup.p2 1.3.0.v20151016-1206, build=1923, branch=8359c49a92734051d767cf7c2d0b8c8d16961b51, commit=8359c49a92734051d767cf7c2d0b8c8d16961b51
Bundle org.eclipse.oomph.setup.pde 1.3.0.v20150923-1022, build=1923, branch=8359c49a92734051d767cf7c2d0b8c8d16961b51, commit=8359c49a92734051d767cf7c2d0b8c8d16961b51
Bundle org.eclipse.oomph.setup.projects 1.3.0.v20150923-0945, build=1923, branch=8359c49a92734051d767cf7c2d0b8c8d16961b51, commit=8359c49a92734051d767cf7c2d0b8c8d16961b51
Bundle org.eclipse.oomph.setup.targlets 1.3.0.v20151118-1556, build=1955, branch=1c21883e4012ea9ca63d5ee8fc0d9a02fc7b4d75, commit=1c21883e4012ea9ca63d5ee8fc0d9a02fc7b4d75
Bundle org.eclipse.oomph.setup.workingsets 1.3.0.v20150923-0945, build=1923, branch=8359c49a92734051d767cf7c2d0b8c8d16961b51, commit=8359c49a92734051d767cf7c2d0b8c8d16961b51
Performing P2 Director (Eclipse IDE for Java Developers (Mars))
Offline = false
Mirrors = true
Resolving 33 requirements from 8 repositories to /home/maggu2810/bin/pkgs/eclipse/smarthome-master/eclipse
Requirement org.eclipse.wst.xml_ui.feature.feature.group
Requirement org.eclipse.m2e.feature.feature.group
Requirement org.eclipse.m2e.logback.feature.feature.group
Requirement org.eclipse.xtend.sdk.feature.group [2.6.0,2.7.0]
Requirement org.eclipse.xtext.sdk.feature.group [2.6.0,2.7.0]
Requirement org.eclipse.emf.mwe2.language.sdk.feature.group [2.6.0,2.7.0]
Requirement org.eclipse.emf.mwe2.runtime.sdk.feature.group [2.6.0,2.7.0]
Requirement org.eclipse.emf.mwe.core.feature.group [1.3.2,1.4.0)
Requirement org.eclipse.emf.mwe.sdk.feature.group [1.3.2,1.4.0)
Requirement org.codehaus.groovy.eclipse.feature.feature.group
Requirement org.sonatype.tycho.m2e.feature.feature.group
Requirement org.eclipse.emf.sdk.feature.group
Requirement org.eclipse.mylyn.java_feature.feature.group
Requirement org.eclipse.mylyn.bugzilla_feature.feature.group
Requirement org.eclipse.mylyn.hudson.feature.group
Requirement org.eclipse.mylyn.git.feature.group
Requirement org.eclipse.pde.feature.group
Requirement org.eclipse.wst.jsdt.feature.feature.group
Requirement org.eclipse.wst.web_ui.feature.feature.group
Requirement epp.package.java [4.5.0,4.6.0)
Requirement org.eclipse.platform.feature.group [4.5.0,4.6.0)
Requirement org.eclipse.rcp.feature.group [4.5.0,4.6.0)
Requirement org.eclipse.jdt.feature.group [3.11.0,3.12.0)
Requirement org.eclipse.oomph.setup.feature.group
Requirement org.eclipse.oomph.targlets.feature.group
Requirement org.eclipse.oomph.setup.targlets.feature.group
Requirement org.eclipse.oomph.setup.jdt.feature.group
Requirement org.eclipse.oomph.setup.git.feature.group
Requirement org.eclipse.egit.feature.group
Requirement org.eclipse.oomph.setup.workingsets.feature.group
Requirement org.eclipse.oomph.setup.projects.feature.group
Requirement org.eclipse.oomph.setup.pde.feature.group
Requirement org.eclipse.oomph.setup.launching.feature.group
Repository http://repo1.maven.org/maven2/.m2e/connectors/m2eclipse-tycho/0.7.0/N/0.7.0.201309291400/
Repository http://download.eclipse.org/modeling/tmf/xtext/updates/composite/releases/
Repository http://download.eclipse.org/modeling/emf/emf/updates/milestones/
Repository http://dist.springsource.org/snapshot/GRECLIPSE/e4.5/
Repository http://download.eclipse.org/technology/epp/packages/mars
Repository http://download.eclipse.org/releases/mars/201510021000
Repository http://download.eclipse.org/oomph/updates/milestone/latest
Repository http://download.eclipse.org/egit/updates
Adding repository http://repo1.maven.org/maven2/.m2e/connectors/m2eclipse-tycho/0.7.0/N/0.7.0.201309291400/
Adding repository http://download.eclipse.org/modeling/tmf/xtext/updates/composite/releases/
Adding repository http://dist.springsource.org/snapshot/GRECLIPSE/e4.5/
Adding repository http://download.eclipse.org/technology/epp/packages/mars
Adding repository http://download.eclipse.org/modeling/emf/emf/updates/milestones/
Adding repository http://download.eclipse.org/oomph/updates/milestone/latest
Adding repository http://download.eclipse.org/releases/mars/201510021000
Adding repository http://download.eclipse.org/egit/updates
Calculating requirements and dependencies.
Cannot complete the request.  Generating details.
ERROR: org.eclipse.equinox.p2.director code=10054 Cannot complete the install because of a conflicting dependency.
  at org.eclipse.oomph.util.OomphPlugin.coreException(OomphPlugin.java:280)
  at org.eclipse.oomph.p2.internal.core.ProfileTransactionImpl.resolve(ProfileTransactionImpl.java:413)
  at org.eclipse.oomph.p2.internal.core.ProfileTransactionImpl.commit(ProfileTransactionImpl.java:325)
  at org.eclipse.oomph.setup.p2.impl.P2TaskImpl.perform(P2TaskImpl.java:710)
  at org.eclipse.oomph.setup.internal.core.SetupTaskPerformer.doPerformNeededSetupTasks(SetupTaskPerformer.java:3025)
  at org.eclipse.oomph.setup.internal.core.SetupTaskPerformer.access$1(SetupTaskPerformer.java:2969)
  at org.eclipse.oomph.setup.internal.core.SetupTaskPerformer$WorkspaceUtil$1.run(SetupTaskPerformer.java:4149)
  at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2241)
  at org.eclipse.oomph.setup.internal.core.SetupTaskPerformer$WorkspaceUtil.performNeededSetupTasks(SetupTaskPerformer.java:4143)
  at org.eclipse.oomph.setup.internal.core.SetupTaskPerformer$WorkspaceUtil.access$0(SetupTaskPerformer.java:4141)
  at org.eclipse.oomph.setup.internal.core.SetupTaskPerformer.performNeededSetupTasks(SetupTaskPerformer.java:2960)
  at org.eclipse.oomph.setup.internal.core.SetupTaskPerformer.performTriggeredSetupTasks(SetupTaskPerformer.java:2935)
  at org.eclipse.oomph.setup.internal.core.SetupTaskPerformer.perform(SetupTaskPerformer.java:2829)
  at org.eclipse.oomph.setup.ui.wizards.ProgressPage$9.run(ProgressPage.java:552)
  at org.eclipse.oomph.setup.ui.wizards.ProgressPage$11$1.run(ProgressPage.java:671)
  at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)
  ERROR: org.eclipse.equinox.p2.director code=0 Software being installed: artificial_root 1.0.0.v1450034371529
  ERROR: org.eclipse.equinox.p2.director code=1 Only one of the following can be installed at once: 
    ERROR: org.eclipse.equinox.p2.director code=0 Modeling Workflow Engine 2 Language 2.5.0.v201312110944 (org.eclipse.emf.mwe2.language 2.5.0.v201312110944)
    ERROR: org.eclipse.equinox.p2.director code=0 Modeling Workflow Engine 2 Language 2.8.0.v201505260456 (org.eclipse.emf.mwe2.language 2.8.0.v201505260456)
    ERROR: org.eclipse.equinox.p2.director code=0 Modeling Workflow Engine 2 Language 2.6.1.v201406120758 (org.eclipse.emf.mwe2.language 2.6.1.v201406120758)
    ERROR: org.eclipse.equinox.p2.director code=0 Modeling Workflow Engine 2 Language 2.5.1.v201402030750 (org.eclipse.emf.mwe2.language 2.5.1.v201402030750)
    ERROR: org.eclipse.equinox.p2.director code=0 Modeling Workflow Engine 2 Language 2.7.0.v201409021051 (org.eclipse.emf.mwe2.language 2.7.0.v201409021051)
    ERROR: org.eclipse.equinox.p2.director code=0 Modeling Workflow Engine 2 Language 2.4.0.v201306110940 (org.eclipse.emf.mwe2.language 2.4.0.v201306110940)
    ERROR: org.eclipse.equinox.p2.director code=0 Modeling Workflow Engine 2 Language 2.4.1.v201309030840 (org.eclipse.emf.mwe2.language 2.4.1.v201309030840)
    ERROR: org.eclipse.equinox.p2.director code=0 Modeling Workflow Engine 2 Language 2.7.1.v201503051215 (org.eclipse.emf.mwe2.language 2.7.1.v201503051215)
    ERROR: org.eclipse.equinox.p2.director code=0 Modeling Workflow Engine 2 Language 2.6.0.v201405200939 (org.eclipse.emf.mwe2.language 2.6.0.v201405200939)
    ERROR: org.eclipse.equinox.p2.director code=0 Modeling Workflow Engine 2 Language 2.8.2.v201511170842 (org.eclipse.emf.mwe2.language 2.8.2.v201511170842)
    ERROR: org.eclipse.equinox.p2.director code=0 Modeling Workflow Engine 2 Language 2.8.1.v201508030411 (org.eclipse.emf.mwe2.language 2.8.1.v201508030411)
    ERROR: org.eclipse.equinox.p2.director code=0 Modeling Workflow Engine 2 Language 2.0.1.v201108020809 (org.eclipse.emf.mwe2.language 2.0.1.v201108020809)
    ERROR: org.eclipse.equinox.p2.director code=0 Modeling Workflow Engine 2 Language 1.0.2.v201102151014 (org.eclipse.emf.mwe2.language 1.0.2.v201102151014)
    ERROR: org.eclipse.equinox.p2.director code=0 Modeling Workflow Engine 2 Language 2.2.0.v201112071250 (org.eclipse.emf.mwe2.language 2.2.0.v201112071250)
    ERROR: org.eclipse.equinox.p2.director code=0 Modeling Workflow Engine 2 Language 2.3.0.v201206120758 (org.eclipse.emf.mwe2.language 2.3.0.v201206120758)
    ERROR: org.eclipse.equinox.p2.director code=0 Modeling Workflow Engine 2 Language 2.1.0.v201111010714 (org.eclipse.emf.mwe2.language 2.1.0.v201111010714)
  ERROR: org.eclipse.equinox.p2.director code=1 Cannot satisfy dependency:
    ERROR: org.eclipse.equinox.p2.director code=0 From: artificial_root 1.0.0.v1450034371529
    ERROR: org.eclipse.equinox.p2.director code=0 To: org.eclipse.emf.mwe2.language.sdk.feature.group [2.6.0,2.7.0]
  ERROR: org.eclipse.equinox.p2.director code=1 Cannot satisfy dependency:
    ERROR: org.eclipse.equinox.p2.director code=0 From: artificial_root 1.0.0.v1450034371529
    ERROR: org.eclipse.equinox.p2.director code=0 To: org.eclipse.emf.mwe2.language.sdk.feature.group [2.8.0,2.9.0)
  ERROR: org.eclipse.equinox.p2.director code=1 Cannot satisfy dependency:
    ERROR: org.eclipse.equinox.p2.director code=0 From: MWE 2 language SDK 2.6.0.v201405200939 (org.eclipse.emf.mwe2.language.sdk.feature.group 2.6.0.v201405200939)
    ERROR: org.eclipse.equinox.p2.director code=0 To: org.eclipse.emf.mwe2.language [2.6.0.v201405200939]
  ERROR: org.eclipse.equinox.p2.director code=1 Cannot satisfy dependency:
    ERROR: org.eclipse.equinox.p2.director code=0 From: MWE 2 language SDK 2.6.1.v201406120758 (org.eclipse.emf.mwe2.language.sdk.feature.group 2.6.1.v201406120758)
    ERROR: org.eclipse.equinox.p2.director code=0 To: org.eclipse.emf.mwe2.language [2.6.1.v201406120758]
  ERROR: org.eclipse.equinox.p2.director code=1 Cannot satisfy dependency:
    ERROR: org.eclipse.equinox.p2.director code=0 From: MWE 2 language SDK 2.8.0.v201505260456 (org.eclipse.emf.mwe2.language.sdk.feature.group 2.8.0.v201505260456)
    ERROR: org.eclipse.equinox.p2.director code=0 To: org.eclipse.emf.mwe2.language [2.8.0.v201505260456]
  ERROR: org.eclipse.equinox.p2.director code=1 Cannot satisfy dependency:
    ERROR: org.eclipse.equinox.p2.director code=0 From: MWE 2 language SDK 2.8.1.v201508030411 (org.eclipse.emf.mwe2.language.sdk.feature.group 2.8.1.v201508030411)
    ERROR: org.eclipse.equinox.p2.director code=0 To: org.eclipse.emf.mwe2.language [2.8.1.v201508030411]
  ERROR: org.eclipse.equinox.p2.director code=1 Cannot satisfy dependency:
    ERROR: org.eclipse.equinox.p2.director code=0 From: MWE 2 language SDK 2.8.2.v201511170842 (org.eclipse.emf.mwe2.language.sdk.feature.group 2.8.2.v201511170842)
    ERROR: org.eclipse.equinox.p2.director code=0 To: org.eclipse.emf.mwe2.language [2.8.2.v201511170842]

Took 39 seconds.
There are failed tasks.
Press Back to choose different settings or Cancel to abort.
```

**Comment by maggu2810** [comment]
After the manual deinstallation of some currently installed features (after the Xtext bump) e.g.
* MWE
* MWE 2
* Xtext
* Xtend
the setup procedure to keep the IDE up to day is working again

**Comment by maggu2810** [comment]
I guess the setup can handle updates of plugins, but no downgrades.

**Comment by kaikreuzer** [comment]
> I guess the setup can handle updates of plugins, but no downgrades.

It is worse, it cannot even handle upgrades - that was my problem with the IDE setup, it complained about conflicting mwe versions... That's why I wanted to revert the setup asap, so that not too many users get the newer xtext plugins and run into problems, when rolling back again.

I guess I will have to dive deeper into Oomph and how to handle IDE updates. The main reason why I pushed for the Xtext update in the first place is that the "check for updates" functionality is always active and automatically updates to Xtext 2.9, which is incompatible to 2.6 in terms of code generation; so whoever accidentially does an automatic IDE upgrade will run into problems.

**Pull request merged**